### ISSUE TITLE
Add polling for CX3550

### ISSUE THREAD

**Issue by kongo09** [kongo09/philips-airpurifier-coap#205]
Some devices need polling to update data as push doesn't work, like CX3550. See discussion here #178 . Idea is to send a beep off value to trigger data polling.

No UI setting is needed as so far this only affects this one device, so the capability setting can be made part of the device properties.

@eulemitkeule

**Comment by EuleMitKeule** [comment]
I'll try to implement this, but it might take a few weeks.

**Comment by iceymanz** [comment]
I've just purchased a CX3550/01 with Firmware: AWS_Philips_AIR_Combo@2.5 device FW 0.1.7

If I run  "aioairctrl --host <device IP> -D status-observe" - on the console I can observe updates made from the app and made from buttons on the device itself.

Although it did crash once when turning on the device with ValueError: non-hexadecimal number found in fromhex() arg at position 488

Can you leverage this polling method? Happy to do any testing / provide logs required.

**Comment by kongo09** [comment]
> I'll try to implement this, but it might take a few weeks.

Any success?

**Comment by EuleMitKeule** [comment]
Nope sorry, didn't really have time and motivation yet. ^^

**Comment by kongo09** [comment]
I'm not really sure if I should keep this open? Do you think you find some motivation to do this, @EuleMitKeule , or shall we close it?

**Comment by EuleMitKeule** [comment]
I mean it's still an ongoing problem and, if you haven't decided otherwise, still an open feature request that anybody could implement and open a PR for, so I don't really see the necessity to close the issue, but that is your decision. I can't promise anything.

**Comment by kongo09** [comment]
Fair enough, I'll keep it open for now

**Comment by Ka3seBr0t** [comment]
Related to [#137](https://github.com/kongo09/philips-airpurifier-coap/issues/137): the CX3550's real protocol is cloud MQTT (AWS IoT device shadow), not local CoAP polling — that's likely why polling needs a workaround here. If anyone wants full native control instead (all functions, verified against physical hardware) rather than the local-CoAP partial support: https://github.com/Ka3seBr0t/HA_Philips_Air_Plus

**Comment by EuleMitKeule** [comment]
@Ka3seBr0t I installed your integration and am able to control the fans, but just like with CoAP, it does not recognize changes made by pressing the physical buttons on the fan.

**Comment by Ka3seBr0t** [comment]
@EuleMitKeule yeah that was a real bug, fixed in v0.3.2. It read the shadow once on connect and then just relied on pushes, but Philips' cloud doesn't reliably push physical button changes so HA went stale. Now there's a 30s shadow/get poll on top, so it resyncs. Tested on a CX3550 here — turned it off at the unit and HA followed by itself.

Update to 0.3.2 in HACS and do a full HA restart (a normal reload won't load the new code), then it should pick up the button presses within ~30s.
